trace/google/cloud/trace/trace_context.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""TraceContext encapsulates the current context within the request's trace."""
+
+from google.cloud.trace.trace import generate_trace_id
+
+import re
+import logging
+
+_TRACE_CONTEXT_HEADER_FORMAT = '([0-9a-f]{32})(\/(\d+))?(;o=(\d+))?'
+
+
+class TraceContext(object):
+    """TraceContext includes 3 fields: traceId, spanId, and an enabled flag 
+    which indicateswhether or not the request is being traced.
```

Comment:
indicateswhether -> indicates whether

TraceContext and ContextTracer and Trace are all very similar names.  Can we make them more distinguishable?  Can we write a sentence describing their relationship to each other?  I think it's like:

Trace: All the spans for a single trace
TraceContext: A textual representation of single span inside a Trace, sent and received in HTTP and RPC messages.  Maybe it should be called SpanSomething instead of TraceSomething? Maybe it should be called TraceHeader?)
ContextTracer: A stack of Spans inside a single Trace.